app/views/shared/_empty_search_state.html.slim

```diff
@@ -12,4 +12,4 @@ tr role="row"
           = t("#{scope}.empty_search.body")
         div class="pf-c-empty-state__primary"
           a class="pf-c-button pf-m-link" type="button" href=url_for
-            = t("#{scope}.empty_search.primary")
```

Comment:
I would change this to `= t('.empty_search.primary')` here, along with changing `en.yml` as follows:

```
  shared:
    empty_search_state:
      empty_search:
        title: No results
        body: There are no items matching your search criteria.
        primary: Clear all filters
 ```
 
Because, as I understand the reason for the `title` and `body` strings here is to provide some defaults in case `scope` is not set. But in the current version, if `scope` is `nil`, `t("#{scope}.empty_search.title")` and `t("#{scope}.empty_search.body")`, would not get resolved properly.

Alternatively, if you keep the current `en.yml`, 
```
  shared:
    empty_search_state:
      title: No results
      body: There are no items matching your search criteria.
      primary: Clear all filters
```

You could do something like this in this partial:
```
- icon = local_assigns[:icon] || :search
- scope = local_assigns[:scope] ? "#{local_assigns[:scope]}.empty_search" : ''

tr role="row"
  td role="cell" colspan="100"
    div class="pf-c-empty-state"
      div class="pf-c-empty-state__content"
        i class="fas fa-#{icon} pf-c-empty-state__icon" aria-hidden="true"
        h1 class="pf-c-title pf-m-lg"
          = t("#{scope}.title")
        div class="pf-c-empty-state__body"
          = t("#{scope}.body")
        div class="pf-c-empty-state__primary"
          a class="pf-c-button pf-m-link" type="button" href=url_for
            = t('.primary')
```

Reply:
Good catch. for the defaults to work, they must follow the same structure in `en.yml` when the scope is say `buyers.accounts.index` as when the scope is `shared.empty_search_state`. IMO the best option is @mayorova's first suggestion, that means adding the `empty_search` and `empty_state` levels.

```yaml
shared:
  empty_search_state:
    empty_search:
      title: No results
      body: There are no items matching your search criteria.
      primary: Clear all filters
```

The second options @mayorova suggests would require updating the value of `scope` in every template calling this partial, which is not desirable.